include/phonemizer.h

```diff
@@ -289,6 +290,42 @@ static const std::map<std::string, std::string> CONTRACTION_PHONEMES = {
 // characters that Espeak-ng treats as stopping tokens.
 static std::string STOPPING_TOKENS = ".,:;!?";
 
+#ifdef ESPEAK_INSTALL
+/**
+ * espeak-ng uses globals to persist and manage its state so it is not compatible with 
+ * threaded parallelism (https://github.com/espeak-ng/espeak-ng/issues/1527).
+ * This singleton acts as a mutex wrapped provider for all espeak phonemization methods such
+ * that multiple instances of the kokoro_runner can be initialized and called in parallel.
+ */
+class espeak_wrapper {
```

Comment:
I'm not really a fan of this approach but I think the issue stems from espeak-ng holding state in globals, so really we are just continuing their pattern by implementing a singleton.